catalog/core/catalog-core-subscriptionstore/subscriptionstore-impl/src/main/java/org/codice/ddf/catalog/subscriptionstore/common/SubscriptionPersistor.java

```diff
@@ -0,0 +1,171 @@
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.subscriptionstore.common;
+
+import static java.lang.String.format;
+import static org.apache.commons.lang.Validate.notEmpty;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionStoreException;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A common persistence layer for all subscriptions targeting the {@link PersistentStore}.
+ * <p>
+ * The {@link SubscriptionPersistor} serves as the data-access object for all {@link ddf.catalog.event.Subscription}s
+ * across any endpoint. The object that actually gets stored is a {@link SubscriptionMetadata}, but combined
+ * with a {@link org.codice.ddf.catalog.subscriptionstore.internal.SubscriptionFactory}, the
+ * original {@code Subscription} object can be re-created and registered with the OSGi framework.
+ * <p>
+ * {@code SubscriptionPersistor} knows how to marshall and unmarshall between {@code SubscriptionMetadata}
+ * and {@link PersistentItem}, effectively hiding the persistence details of subscriptions from the rest
+ * of the service.
+ */
+public class SubscriptionPersistor {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SubscriptionPersistor.class);
+
+    private static final String SUBSCRIPTION_ID_SOLR = "id";
+
+    private static final String SUBSCRIPTION_TYPE = "type";
+
+    private static final String SUBSCRIPTION_SERIALIZED_FILTER = "filter";
+
+    private static final String SUBSCRIPTION_CALLBACK_URL = "callback";
+
+    private final PersistentStore persistentStore;
+
+    public SubscriptionPersistor(PersistentStore persistentStore) {
+        this.persistentStore = persistentStore;
+    }
+
+    /**
+     * Get all the subscription metadata objects in the persistent store, regardless of type.
+     *
+     * @return a map of subscription id's to their accompanying metadata.
+     * @throws SubscriptionStoreException if a problem occurs during the read.
+     */
+    public Map<String, SubscriptionMetadata> getSubscriptions() {
+        List<Map<String, Object>> results;
+        try {
+            results = persistentStore.get(PersistentStore.EVENT_SUBSCRIPTIONS_TYPE);
+        } catch (PersistenceException e) {
+            throw new SubscriptionStoreException("Exception while reading subscriptions from Solr: ",
+                    e);
+        }
+
+        return results.stream()
+                .map(this::formatIfPersistentItem)
+                .map(this::mapToMetadata)
```

Comment:
Validation has been removed from the model. It will appear in PR 3/4 against all traffic to enter the main service impl.